test/controllers/user_controller_test.exs

```diff
@@ -0,0 +1,122 @@
+defmodule Designator.UserControllerTest do
+  use Designator.ConnCase
+
+  @username Application.fetch_env!(:designator, :api_auth)[:username]
+  @password Application.fetch_env!(:designator, :api_auth)[:password]
+
+  setup %{conn: conn} do
+    workflow_id = 2
+    user_id = 1
+    user_cache_key = { workflow_id, user_id }
+    # initialize / reset the user cache data
+    # couldn't get the cache store to reset as per
+    # https://github.com/sasa1977/con_cache#testing-in-your-application
+    Designator.UserCache.set(
+      user_cache_key,
+      %{
+        seen_ids: MapSet.new([]),
+        recently_selected_ids: MapSet.new,
+        configuration: %{}
+      }
+    )
+    user = Designator.UserCache.get(user_cache_key)
+    {
+      :ok,
+      [
+        user: user,
+        workflow_id: 2,
+        conn: put_req_header(conn, "accept", "application/json")
+      ]
+    }
+  end
+
+  def add_seens_path(user_id) do
+    "/api/users/#{user_id}/add_seen_subjects"
+  end
+
+  def put_req(conn, user_id, workflow_id, subject_ids) do
+    params = Poison.encode!(%{workflow_id: workflow_id, subject_ids: subject_ids})
+    conn
+    |> put_req_header("content-type", "application/json")
```

Comment:
Required because of PUT vs. POST? Workflow controller POST methods don't need it.

Reply:
I don't think the previous tests use what i've done but the internal request routing and params mapping works ok without it. I'd really like to mirror the payloads we're going to send in https://github.com/zooniverse/Panoptes/pull/3150

If we don't encode the params map using `Poison` we end up with the params map being used to setup the content-type header, i.e. not json
```
req_headers: [
    {"accept", "application/json"},
    {"authorization", "Basic dXNlcm5hbWU6cGFzc3dvcmQ="}, 
    {"content-type", "multipart/mixed; charset: utf-8"}
]
```
I've used Poison to encode the data to a string json representation before sending and applied the `content-type` header as well as per https://github.com/zooniverse/Panoptes/blob/02178155915f7d8b748175d139c569333f12d8c4/app/services/designator_client.rb#L65

E.g. with encoded data and header set
```
req_headers: [
    {"accept", "application/json"},
    {"content-type", "application/json"},
    {"authorization", "Basic dXNlcm5hbWU6cGFzc3dvcmQ="}
]
```

Reply:
Using Poison to encode the params makes sense, I was wondering just about that header. But 

>I'd really like to mirror the payloads we're going to send in zooniverse/Panoptes#3150

makes sense, best to test what we'll actually be sending if Panoptes includes the extra header by default.

Reply:
without the header the specs fail with `** (ArgumentError) a content-type header is required when setting a binary body in a test connection`